lib/ide/src/executor/web.rs

```diff
@@ -0,0 +1,104 @@
+//! Module defining `JsExecutor` - an executor that tries running until stalled
+//! on each animation frame callback call.
+
+use crate::prelude::*;
+
+use basegl::control::callback::CallbackHandle;
+use basegl::control::EventLoopCallback;
+use basegl::control::EventLoop;
+use futures::task::LocalSpawn;
+use futures::task::LocalFutureObj;
+use futures::task::SpawnError;
+use futures::executor::LocalPool;
+use futures::executor::LocalSpawner;
+
+/// Executor. Uses a single-threaded `LocalPool` underneath, relying on basegl's
+/// `EventLoop` to do as much progress as possible on every animation frame.
+#[derive(Debug)]
+pub struct JSExecutor {
+    /// Underlying executor. Shared internally with the event loop callback.
+    executor    : Rc<RefCell<LocalPool>>,
```

Comment:
Could we make a newtype for `Rc<RefCell<LocalPool>>`? We should make RefCells accessible in the smallest possible amount of places, so making a small, self-contained shared pool is better than this.

Reply:
See the answer to the comment about borrow_mut usage below.